server/client.ts

```diff
@@ -465,46 +465,54 @@ class Client {
 		const cmd = args?.shift()?.toLowerCase() || "";
 
 		const irc = target.network.irc;
-		let connected = irc && irc.connection && irc.connection.connected;
+		const connected = irc?.connected;
+
+		const emitFailureDisconnected = () => {
+			target.chan.pushMessage(
+				this,
+				new Msg({
+					type: MessageType.ERROR,
+					text: "You are not connected to the IRC network, unable to send your command.",
+				})
+			);
+		};
 
 		if (inputs.userInputs.has(cmd)) {
-			const plugin = inputs.userInputs.get(cmd);
+			const plugin = inputs.userInputs.get(cmd)!; // this is only save because we just checked it
```

Comment:
nit: I meant to suggest changes here, but I'd prefer something like https://github.com/thelounge/thelounge/pull/4714

Reply:
your PR over there fails tze typing check though

Reply:
done